kong/sni.go

```diff
@@ -0,0 +1,23 @@
+package kong
+
+// SNI represents a SNI in Kong.
+// Read https://getkong.org/docs/0.14.x/admin-api/#sni-object
+// +k8s:deepcopy-gen=true
+type SNI struct {
+	ID          *string      `json:"id,omitempty" yaml:"id,omitempty"`
+	Name        *string      `json:"name,omitempty" yaml:"name,omitempty"`
+	CreatedAt   *int64       `json:"created_at,omitempty" yaml:"created_at,omitempty"`
+	Certificate *Certificate `json:"certificate,omitempty" yaml:"certificate,omitempty"`
+	Tags        []*string    `json:"tags,omitempty" yaml:"tags,omitempty"`
+}
+
+// FriendlyName returns the endpoint key name or ID.
```

Comment:
I believe this comment is copy pasted to each type not necessarily describing each of the respective `FriendlyName()`s.

Is "returns the endpoint key name or ID" accurate for e.g. SNI? 🤔

Reply:
This is all existing code (as indicated by the commit/PR description), the only thing I actually changed was this:
https://github.com/Kong/go-kong/pull/281/commits/e2a20300f0a2c55c84621786f0f512db68972add

We can review these things if we want, but I wanted to keep this PR as simple as possible given how much refactoring it's already doing.

Reply:
Fair enough. Let's leave it as is and tackle those issues separately. 👍

Reply:
Great, thanks @pmalek. 👍